cmd/cli/populate.go

```diff
@@ -0,0 +1,989 @@
+}
+
+func DoStartPollTx(address crypto.AddressI) {
+	// log the tx
+	fmt.Printf("Executing start poll tx for %s\n", address)
+	// send the tx
+	tx := &txStartPoll{
+		Fee:      20_000,
+		PollJSON: []byte(`{"proposal": "canopy network is the best","endBlock": 100,"URL": "https://discord.com/link-to-thread"}`),
+		Submit:   true,
+		Password: Pwd,
+		fromFields: fromFields{
+			Address: address.Bytes(),
+		},
+	}
+	postTx(tx, TxStartPollTxURL)
+}
+
+func DoVotePollTx(address crypto.AddressI) {
+	// log the tx
+	fmt.Printf("Executing vote poll tx for %s\n", address)
+	// send the tx
+	tx := &txVotePoll{
+		Fee:         20_000,
+		PollJSON:    []byte(`{"proposal": "canopy network is the best","endBlock": 100,"URL": "tcp://discord.com/link-to-thread"}`),
+		PollApprove: true,
+		Submit:      true,
+		Password:    Pwd,
+		fromFields: fromFields{
+			Address: address.Bytes(),
+		},
+	}
+	postTx(tx, TxVotePollTxURL)
+}
+
+func WaitBlocks(numBlocks ...int) {
+	waitBlocks, lastHeight := 2, 0
+	type HeightResp struct {
+		Height int `json:"height"`
+	}
+	if len(numBlocks) == 1 {
+		waitBlocks = numBlocks[0]
+	}
+	for range time.Tick(time.Second) {
+		// get height
+		got, err := post(QueryHeightURL, nil)
+		if err != nil {
+			panic(err)
+		}
+		// unmarshal response
+		resp := &HeightResp{}
+		if err = json.Unmarshal(got, resp); err != nil {
+			panic(err)
+		}
+		if lastHeight != 0 && resp.Height >= lastHeight+waitBlocks {
+			return
+		}
+		if lastHeight == 0 {
+			lastHeight = resp.Height
+		}
+	}
+}
+
+var count int
+
+func postTx(obj any, url string) string {
+	count++
+	// marshal the tx
+	bz, e := json.MarshalIndent(obj, "", "  ")
+	if e != nil {
+		log.Fatal("Error marshalling:", e)
+	}
+	// send the txn
+	hash, e := post(url, bz)
+	if e != nil {
+		log.Fatal("Error POSTing transfer:", e)
+	}
+	fmt.Printf("Tx #%d, Hash:%s\n", count, string(hash))
+	return strings.Trim(string(hash), "\"")
+}
+
+var httpClient = &http.Client{}
+
+func post(url string, bz []byte) ([]byte, error) {
+	// generate the request
+	request, e := http.NewRequest("POST", url, bytes.NewBuffer(bz))
+	if e != nil {
+		return nil, fmt.Errorf("Error creating request to %s:%s\n", url, e.Error())
+	}
+	// execute the request
+	resp, e := httpClient.Do(request)
+	if e != nil {
+		return nil, fmt.Errorf("Error executing request to %s:%s\n", url, e.Error())
+	}
+	defer resp.Body.Close()
+	// check the status code
+	if resp.StatusCode != 200 {
+		return nil, fmt.Errorf("Non 200 status code from %s:%d\n", url, resp.StatusCode)
+	}
+	// read the request bytes
+	respBz, e := io.ReadAll(resp.Body)
+	if e != nil {
+		return nil, fmt.Errorf("Error reading response from %s:%s\n", url, e.Error())
+	}
+	// return
+	return respBz, nil
+}
+
+// =====================================================
+// Query Request Types
+// =====================================================
+type heightRequest struct {
+	Height uint64 `json:"height"`
+}
+
+type chainRequest struct {
```

Comment:
I'll only use this struct as an example but there's like ~10 `<type>Request` structs unused on this populator, we might want to clean them up if they wont be ultimately used